### DISCUSSION TITLE
Proper use of PDM docker image in Dockerfile?

### DISCUSSION THREAD

**Discussion by Borketh** [pdm-project/pdm#957] [Q&A]
I am trying to split my docker build into multiple stages. I am aware of a docker image for PDM, so I have two questions:
Is this a supported method of using PDM?
How would I best use it this way?

My envisioned steps are thus

- Gather dependencies
    - Use PDM image or `curl installer | python` 
    - copy toml, lock, and readme
    - install deps specified in toml

- Install project
    - get deps folder from prev stage
    - install all three packages

- run 
    - get deps folder from prev stage (now has my three packages)
    - `python3 -m project`

```dockerfile
# work in progress!
FROM frostming/pdm

# folders for image
VOLUME /config
WORKDIR /build

RUN echo $(python3 -V)
COPY pyproject.toml *.lock *.env README* ./
RUN pdm sync -v

COPY project package1 package2 /build/
RUN pdm install -v --prod --no-editable

# start the application

# FROM python:3.10 AS RUN
COPY --from=INSTALL /build/__pypackages__/3.10/lib /allpkgs/
ENV PYTHONPATH=/allpkgs/
CMD python3 -m project
```

I have tried using the image as-is, but it says it couldn't find a compatible python (with ~=3.10)

**Comment by Borketh** [comment]
Addendum: eventually one or more of the packages will be Cython, with luck

**Comment by pawamoy** [comment]
Something like this? https://pdm.fming.dev/usage/advanced/#use-pdm-in-a-multi-stage-dockerfile

**Reply by Borketh** [reply]
I already looked at that, but I meant use of pdm from the recommended installer, or through the [docker image](https://hub.docker.com/r/frostming/pdm).

**Reply by simonprovost** [reply]
I saw your Github issue @Borketh and believed it would be feasible to do this today, however the docker image was updated three years ago, thus I am assuming this is no longer functional. 

However, @pawamoy's suggestion is a great starting point! Solved all my issues. The documentation about how PDM and Docker IMHO could be further enhanced but it does the minimal job at least which is good!

Happy PDM!
